src/earn/EarnEnterAmount.tsx

```diff
@@ -202,30 +234,39 @@ function EarnEnterAmount({ route }: Props) {
         localAmount: parsedLocalAmount,
       }
     }
-  }, [tokenAmountInput, localAmountInput, enteredIn, token])
+  }, [tokenAmountInput, localAmountInput, enteredIn, transactionToken, mode])
 
-  const feeCurrencies = useSelector((state) => feeCurrenciesSelector(state, token.networkId))
+  const feeCurrencies = useSelector((state) =>
+    feeCurrenciesSelector(state, transactionToken.networkId)
+  )
 
   useEffect(() => {
     clearPreparedTransactions()
 
     if (
       !tokenAmount ||
       tokenAmount.isLessThanOrEqualTo(0) ||
-      tokenAmount.isGreaterThan(token.balance)
+      tokenAmount.isLessThanOrEqualTo(0) ||
+      tokenAmount.isGreaterThan(
+        isWithdrawal
+          ? transactionToken.balance.multipliedBy(pool.pricePerShare[0])
+          : transactionToken.balance
+      )
     ) {
       return
     }
     const debouncedRefreshTransactions = setTimeout(() => {
-      return handleRefreshPreparedTransactions(tokenAmount, token, feeCurrencies)
+      return handleRefreshPreparedTransactions(tokenAmount, transactionToken, feeCurrencies)
```

Comment:
for withdraw, shouldn't we convert the `tokenAmount` (which is in units of the `depositToken`) to units of `transactionToken` / `withdrawToken` before we prepare the transaction?

Reply:
Fixed in 24c47be2e5d250b88f07fba17dee96cbebe7d566!